src/main/webapp/app/course/dashboards/assessment-dashboard/assessment-dashboard.component.ts

```diff
@@ -120,9 +122,42 @@ export class AssessmentDashboardComponent implements OnInit, AfterViewInit {
                 });
 
                 this.extractExercises(exercises);
-
-                // TODO: implement some tutor stats here similar to the ones below but based on the exam and not the course
             });
+            this.examManagementService.getStatsForExamAssessmentDashboard(this.courseId, this.examId).subscribe(
+                (res: HttpResponse<StatsForDashboard>) => {
+                    this.stats = StatsForDashboard.from(res.body!);
+                    this.numberOfSubmissions = this.stats.numberOfSubmissions;
+                    this.numberOfAssessmentsOfCorrectionRounds = this.stats.numberOfAssessmentsOfCorrectionRounds;
+
+                    this.totalNumberOfAssessments = new DueDateStat();
+                    for (const dueDateStat of this.numberOfAssessmentsOfCorrectionRounds) {
+                        this.totalNumberOfAssessments.inTime += dueDateStat.inTime;
+                    }
+                    this.numberOfCorrectionRounds = this.numberOfAssessmentsOfCorrectionRounds.length;
+
+                    this.numberOfComplaints = this.stats.numberOfComplaints;
+                    this.numberOfOpenComplaints = this.stats.numberOfOpenComplaints;
+                    this.numberOfAssessmentLocks = this.stats.numberOfAssessmentLocks;
+                    this.totalNumberOfAssessmentLocks = this.stats.totalNumberOfAssessmentLocks;
+
+                    // the received leaderboard from the server is still empty. TODO: fill on server side
+                    const tutorLeaderboardEntry = this.stats.tutorLeaderboardEntries.find((entry) => entry.userId === this.tutor.id);
+                    this.sortService.sortByProperty(this.stats.tutorLeaderboardEntries, 'points', false);
+                    if (tutorLeaderboardEntry) {
+                        this.numberOfTutorAssessments = tutorLeaderboardEntry.numberOfAssessments;
+                        this.numberOfTutorComplaints = tutorLeaderboardEntry.numberOfTutorComplaints;
+                    } else {
+                        this.numberOfTutorAssessments = 0;
+                        this.numberOfTutorComplaints = 0;
+                    }
+
+                    if (this.numberOfSubmissions.total > 0) {
+                        this.totalAssessmentPercentage = Math.floor((this.totalNumberOfAssessments.total / (this.numberOfSubmissions.total * this.numberOfCorrectionRounds)) * 100);
```

Comment:
Shouldn't you reset this if else?

Reply:
I guess theoretically you could reset it in an else case, but practically it will never get to this case, as you can't decrease the number of submissions inside this component. And when the component is reloaded and the default value is 0 again, in case the numberOfSubmissions is 0 too.